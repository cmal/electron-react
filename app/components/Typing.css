.typing {
  height: 100%;
  width: 90%;
}

legend {
  display: none;
}

fieldset {
  border-color: rgba(255, 255, 255, 0.2) !important;
}

.typingOriginText {
  height: 40%;
}

.typingOriginText div {
  height: 100%;
}

.typingOriginText textarea {
  height: 100%;
  /* overflow: scroll; */
  color: rgba(255, 255, 255, 0.7) !important;
}

.typingText {
  height: 30%;
}

.typingText div {
  height: 100%;
}

.typingText textarea {
  height: 100%;
  /* overflow: scroll; */
  color: rgba(255, 255, 255, 0.7) !important;
}
