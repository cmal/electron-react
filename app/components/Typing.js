// @flow
import React, { Component } from 'react';
// import { Link } from 'react-router-dom';
import TextField from '@material-ui/core/TextField';
import styles from './Typing.css';
// import routes from '../constants/routes.json';
import type { TypingState } from '../reducers/types';
import { typingState } from '../reducers/types';

export default class Typing extends Component<TypingState> {
  props: TypingState;

  constructor(props: TypingState) {
    super(props);

    this.state = typingState;

    this.handleChange = event => {
      console.log('change');
      console.log(event.target.value);
      this.setState({
        typingText: event.target.value
      });
    };

    this.handleCompositionUpdate = event => {
      console.log('update');
      console.log(event.target.value);
    };

    this.handlComositionEnd = event => {
      console.log('end');
      console.log(event.target.value);
    };

    this.handleCompositionStart = event => {
      console.log('start');
      console.log(event.target.value);
    };

    this.handleKeyDown = event => {
      console.log(event);
    };
  }

  state: TypingState = typingState;

  // getInitialState = () => typingState;

  render() {
    const { typing, loadClipboard } = this.props;
    const { text, typingText } = this.state;
    return (
      <div className={styles.typing}>
        <TextField
          id="typing-text"
          label=""
          style={{ margin: 8 }}
          placeholder=""
          helperText=""
          fullWidth
          disabled
          multiline
          margin="normal"
          value={typing.text}
          text={text}
          className={styles.typingOriginText}
          variant="outlined"
          InputLabelProps={{
            shrink: true
          }}
        />
        <div className={styles.btnGroup}>
          <button
            className={styles.btn}
            onClick={loadClipboard}
            data-tclass="btn"
            type="button"
          >
            LOAD CLIPBOARD
          </button>
        </div>
        <TextField
          id="typing-text"
          label=""
          style={{ margin: 8 }}
          placeholder="xxx"
          helperText=""
          fullWidth
          multiline
          margin="normal"
          value={typing.typingText}
          typingText={typingText}
          className={styles.typingText}
          variant="outlined"
          onChange={this.handleChange}
          onCompositionUpdate={this.handleCompositionUpdate}
          onCompositionEnd={this.handleCompositionEnd}
          onCompositionStart={this.handleCompositionStart}
          onKeyDown={this.handleKeyDown}
          InputLabelProps={{
            shrink: true
          }}
        />
      </div>
    );
  }
}
