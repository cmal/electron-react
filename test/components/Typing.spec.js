import { spy } from 'sinon';
import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Typing from '../../app/components/Typing';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
  const actions = {
    loadClipboard: spy()
  };
  const component = shallow(
    <Typing typing={{ text: 'demo', typingText: '' }} {...actions} />
  );
  return {
    component,
    actions
  };
}

describe('Typing component', () => {
  it('should have initial state', () => {
    const { component } = setup();
    expect(component.find('.typingOriginText').at(0)).toHaveProperty(
      ['typing', 'text'],
      'demo'
    );
  });

  it('should display text', () => {
    const { component } = setup();
    expect(component.find('.typingOriginText textarea').at(0)).toHaveProperty(
      ['typing', 'text'],
      'demo'
    );
  });

  it('should display typingText', () => {
    const { component } = setup();
    expect(component.find('.typingText textarea').at(0)).toHaveProperty(
      ['typing', 'typingText'],
      ''
    );
  });

  it('should first button should call increment', () => {
    const { component, actions } = setup();
    component
      .find('button')
      .at(0)
      .simulate('click');
    expect(actions.loadClipboard.called).toBe(true);
  });

  it('should match exact snapshot', () => {
    const { actions } = setup();
    const typing = (
      <div>
        <Router>
          <Typing typing={{ text: 'demo', typingText: '' }} {...actions} />
        </Router>
      </div>
    );
    const tree = renderer.create(typing).toJSON();

    expect(tree).toMatchSnapshot();
  });

  // it('should second button should call decrement', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(1).simulate('click');
  //   expect(actions.decrement.called).toBe(true);
  // });

  // it('should third button should call incrementIfOdd', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(2).simulate('click');
  //   expect(actions.incrementIfOdd.called).toBe(true);
  // });

  // it('should fourth button should call incrementAsync', () => {
  //   const { buttons, actions } = setup();
  //   buttons.at(3).simulate('click');
  //   expect(actions.incrementAsync.called).toBe(true);
  // });
});
