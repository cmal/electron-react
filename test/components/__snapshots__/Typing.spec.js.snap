// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typing component should match exact snapshot 1`] = `
<div>
  <div
    className="typing"
  >
    <div
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4 typingOriginText"
      style={
        Object {
          "margin": 8,
        }
      }
    >
      
      <div
        className="MuiInputBase-root-18 MuiOutlinedInput-root-5 MuiInputBase-disabled-21 MuiOutlinedInput-disabled-7 MuiInputBase-fullWidth-27 MuiInputBase-formControl-19 MuiInputBase-multiline-26 MuiOutlinedInput-multiline-11"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="MuiNotchedOutline-root-35 MuiNotchedOutline-disabled-39 MuiOutlinedInput-notchedOutline-12"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="MuiNotchedOutline-legend-36"
            style={
              Object {
                "width": 8,
              }
            }
          />
        </fieldset>
        <div
          className="Textarea-root-40"
        >
          <textarea
            aria-hidden="true"
            className="Textarea-textarea-41 Textarea-shadow-42"
            readOnly={true}
            rows="1"
            tabIndex={-1}
            value=""
          />
          <textarea
            aria-hidden="true"
            className="Textarea-textarea-41 Textarea-shadow-42"
            readOnly={true}
            rows={1}
            tabIndex={-1}
            value="Helper Text"
          />
          <textarea
            aria-invalid={false}
            className="Textarea-textarea-41 MuiInputBase-input-28 MuiOutlinedInput-input-13 MuiInputBase-disabled-21 MuiOutlinedInput-disabled-7 MuiInputBase-inputMultiline-30 MuiOutlinedInput-inputMultiline-15"
            disabled={true}
            id="typing-text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder=""
            required={false}
            rows={1}
            style={
              Object {
                "height": 19,
              }
            }
            value="Helper Text"
          />
        </div>
      </div>
      
    </div>
    <div
      className="btnGroup"
    >
      <button
        className="btn"
        data-tclass="btn"
        onClick={[Function]}
        type="button"
      >
        LOAD CLIPBOARD
      </button>
    </div>
    <div
      className="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4 typingText"
      onCompositionStart={[Function]}
      onCompositionUpdate={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "margin": 8,
        }
      }
    >
      
      <div
        className="MuiInputBase-root-18 MuiOutlinedInput-root-5 MuiInputBase-fullWidth-27 MuiInputBase-formControl-19 MuiInputBase-multiline-26 MuiOutlinedInput-multiline-11"
        onClick={[Function]}
      >
        <fieldset
          aria-hidden={true}
          className="MuiNotchedOutline-root-35 MuiOutlinedInput-notchedOutline-12"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="MuiNotchedOutline-legend-36"
            style={
              Object {
                "width": 8,
              }
            }
          />
        </fieldset>
        <div
          className="Textarea-root-40"
        >
          <textarea
            aria-hidden="true"
            className="Textarea-textarea-41 Textarea-shadow-42"
            readOnly={true}
            rows="1"
            tabIndex={-1}
            value=""
          />
          <textarea
            aria-hidden="true"
            className="Textarea-textarea-41 Textarea-shadow-42"
            readOnly={true}
            rows={1}
            tabIndex={-1}
            value=""
          />
          <textarea
            aria-invalid={false}
            className="Textarea-textarea-41 MuiInputBase-input-28 MuiOutlinedInput-input-13 MuiInputBase-inputMultiline-30 MuiOutlinedInput-inputMultiline-15"
            disabled={false}
            id="typing-text"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="xxx"
            required={false}
            rows={1}
            style={
              Object {
                "height": 19,
              }
            }
            value=""
          />
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
